Build/QIR_validation.ps1

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+$ErrorActionPreference = 'Stop'
+
+& "$PSScriptRoot/set-env.ps1"
+$all_ok = $True
+
+function Build-One {
+    param(
+        $project
+    );
+
+    $project_directory = $(Split-Path $project -Parent)
+    $project_name = $(Split-Path $project -LeafBase)
+
+    pushd $project_directory
+
+    Write-Host "##[info]Building $project..."
+    dotnet build $project `
+        -c $Env:BUILD_CONFIGURATION `
+        -v $Env:BUILD_VERBOSITY `
+        --no-dependencies `
+        /property:DefineConstants=$Env:ASSEMBLY_CONSTANTS `
+        /property:Version=$Env:ASSEMBLY_VERSION `
+        /property:QirGeneration="true"
+
+    if  ($LastExitCode -ne 0) {
+        Write-Host "##vso[task.logissue type=error;]Failed to build $project."
+        $script:all_ok = $False
+    } else {
+        qir-cli `
+            --dll (Join-Path bin $Env:BUILD_CONFIGURATION netcoreapp3.1 "${project_name}.dll") `
+            --exe qir
+        if  ($LastExitCode -ne 0) {
+            Write-Host "##vso[task.logissue type=error;]Failed to build $project."
+            $script:all_ok = $False
+        } else {
+            Get-ChildItem (Join-Path qir *__Interop.exe) `
+                | ForEach-Object { & $_ }
+            if  ($LastExitCode -ne 0) {
+                Write-Host "##vso[task.logissue type=error;]$project encountered an error or failed during execution."
+                $script:all_ok = $False
+            }
+        }
+    }
+
+    if ($env:FORCE_CLEANUP -eq "true") {
+        # Force cleanup of generated bin, obj, and qir folders for this project.
+        Write-Host "##[info]Cleaning up bin/obj from $project_directory..."
+        Get-ChildItem -Path $project_directory -Recurse | Where-Object { ($_.name -eq "bin" -or $_.name -eq "obj" -or $_.name -eq "qir") -and $_.attributes -eq "Directory" } | Remove-Item -recurse -force
+    }
+
+    popd
+}
+
+# The commented out lines are sample projects that are not yet compatible for QIR generation/execution. 
+# 'not compatible' means that the structure of the sample is not compatible for QIR generation.
+# 'needs argument(s)' means that the sample can generated QIR, but running the exe requires command line arguments.
```

Comment:
There are quite a lot of samples that can be compiled with QIR, but not executed. I had wanted to find a way to add some switch logic to the script that allowed me to specify per sample how far I wanted to go with the validation: build dll, build executables, run executables. Depending on the current blockers for a given sample, it may not be able to be run, but it can still be useful to compile to tests that part of the QIR compilation process, for example. Have a switch like this would also make it easier to onboard samples that are currently not compatible as they could be onboarded in stages.

I don't have much experience with powershell, so I wasn't able to figure out how to implement this switch logic, but maybe someone with more experience can show me how something like that would look.

Reply:
It would be great to figure out how to run samples that need arguments in a programmatic way. I guess the alternative would be to hardcode the needed arguments for each sample that needs them. If that is the direction to go it, I would need some help with the powershell in learning how to do that.

Reply:
Let's focus on having everything that builds also be able to run - it seems that's more of a setup issue and it should never be the case that something that builds can't run.

Reply:
sounds good to me